### PR TITLE
🐛✨Avoid disabling buttons on submit. Send clicked submit buttons with form submit

### LINKED ISSUE #16461: Multiple submit buttons in amp-form
## What's the issue?

-   Using `<form>` with the `action-xhr` attribute ignores `name` (and `value`) of `<input type="submit">`
-   `formaction` attribute is not allowed

This makes it difficult to build forms with more than one submit button (one example is a dialog with a "Yes"/"No" question where both options should result in a POST).

## How do we reproduce the issue?

```html
<form method="get" action="submit" target="_top">
  <fieldset>
    <input type="hidden" name="id" value="abc">
    <input type="submit" name="decision" value="Yes">
    <input type="submit" name="decision" value="No">
  </fieldset>
</form>
```
Pressing "Yes" correctly makes a request to `submit?id=abc&decision=Yes`

```html
<form method="get" action-xhr="submit">
  <fieldset>
    <input type="hidden" name="id" value="abc">
    <input type="submit" name="decision" value="Yes">
    <input type="submit" name="decision" value="No">
  </fieldset>
</form>
```

Pressing "Yes" makes a request to `submit?id=abc` (the `decision` parameter is not present). Same applies for POST forms.

## Which AMP version is affected?

AMP version 1530043289240

## Proposal

I think the discrepancy between `action` and `action-xhr` can be considered a bug, so perhaps short-term just disallowing the `name` attribute on `<input type="submit">` makes sense.

That said, would be good to allow for multiple submit buttons, since that would make some use-cases easier to do with forms, e.g. rating and prompts to approve / reject things.

### LINKED ISSUE #18285: amp-form: "clear" action resets button "disabled" state
If the button has a `[disabled]` binding, it can become out of sync.

Looks like this was added a long time ago to prevent duplicate form submissions: https://github.com/ampproject/amphtml/pull/3582. Perhaps we can change that to a programmatic check to avoid mucking with the disabled attribute. 

`b/115780963`

### LINKED ISSUE #18745: getFormAsObject doesn't include <button> with name and value
When submitting a form normally, if the form includes a `<button>` with `name` and `value` attributes, the name-value pair will actually be included in the form submission query string (or `multipart/form- data`). However, `getFormAsObject` doesn't seem to include <button> with name and value, which is not consistent with browser's default form submission behavior.

### PR DESCRIPTION
**Pull request by cvializ**
Closes #18745 
Closes #16461 
Closes #18285

This fix removes the button-disabling behavior, which had little effect anyway since there is a submit handler that prevents submit if the form is currently submitting. Publishers who need to disable their submit buttons on submit can do so with amp-bind.

This PR also adds the clicked submit button value to the request body. The solution here adds the focused submit button to the form data. There are not many good ways to detect which button was used to trigger submit, and the spec provides few details about when a submit button must be included in the request. Using the activeElement property seemed to be the lowest-overhead.

/to @aghassemi
/cc @choumx

### CODE REVIEW

**Comment by aghassemi** [comment]
🗿 🐦 🐦 🐦  😮

**Comment by cvializ** [comment]
Some non-hermetic tests are broken by this PR's tests somehow. Investigating
https://travis-ci.org/ampproject/amphtml/jobs/443229291#L785

**Comment by cvializ** [comment]
@aghassemi I fixed the tests, PTAL

**Comment by cvializ** [comment]
Hmm I see what you mean, but I see it more as a behavior change than as a change of something in the documentation that people were told to depend on. If it had been a public CSS like `.amp-submit-disabled` class I would agree with the caveat more. Do you mean like someone could have used CSS rules like `button[disabled] ~ .wait-message`?

File: src/form.js [hunk 1]
```diff
@@ -40,38 +40,61 @@ export function setFormForElement(element, form) {
...
         if (option.selected) {
           data[name].push(option.value);
         }
       });
-    } else {
-      data[name].push(input.value);
+      continue;
     }
+
+    // Browsers vary on the conditions that focus the form submit button.
```
Comment (aghassemi) [inline src/form.js]: add maybe add "In a form, the button's value is only submitted if the button itself was used to submit the form."

**Pull request merged**